### PR TITLE
FEATURE: Adds a condition variable for trigger controlling

### PR DESCRIPTION
**Pull request by Stefan-Kosker**

### CODE REVIEW

File: Classes/Archivist.php [hunk 1]
```diff
@@ -62,6 +63,13 @@ class Archivist
      */
     public function organizeNode(NodeInterface $triggeringNode, array $sortingInstructions)
     {
+        if (isset($sortingInstructions['condition'])) {
+            $condition = $this->eelEvaluationService->evaluate($sortingInstructions['condition'], ['node' => $triggeringNode]);
+            if ($condition === null) {
```
Comment (daniellienert) [inline Classes/Archivist.php]: !== true

File: Configuration/Testing/Settings.yaml [hunk 2]
```diff
@@ -16,6 +16,9 @@ PunktDe:
         # or an eel expression like seen below
         sorting: title
 
+        # Optional: Trigger sorting only, when condition is met.
+        condition: "${node.properties.date}"
```
Comment (daniellienert) [inline Configuration/Testing/Settings.yaml]: condition: "${node.properties.date != null}"

File: Tests/Functional/ArchivistTest.php [hunk 3]
```diff
@@ -177,4 +192,12 @@ protected function createNode($nodeName = 'trigger-node', array $properties = []
 
         return $this->node->createNodeFromTemplate($triggerNodeTemplate, $nodeName);
     }
+
+    protected function setNodeSettings($node, array $properties = [])
```
Comment (daniellienert) [inline Tests/Functional/ArchivistTest.php]: Can be removed

File: Readme.md [hunk 4]
```diff
@@ -36,6 +36,9 @@ PunktDe:
         # or an eel expression like seen below
         sorting: title
 
+        # Optional: Trigger sorting only, when condition is met.
```
Comment (daniellienert) [inline Readme.md]: # Optional: Trigger sorting only, when condition is met. Can be used to make sure that required properties are set as expected.

**Review by daniellienert — CHANGES_REQUESTED** [review]

File: Classes/Archivist.php (hunk 1 above)
Comment (Stefan-Kosker) [reply Classes/Archivist.php]: done

File: Configuration/Testing/Settings.yaml (hunk 2 above)
Comment (Stefan-Kosker) [reply Configuration/Testing/Settings.yaml]: done

File: Tests/Functional/ArchivistTest.php (hunk 3 above)
Comment (Stefan-Kosker) [reply Tests/Functional/ArchivistTest.php]: done

File: Readme.md (hunk 4 above)
Comment (Stefan-Kosker) [reply Readme.md]: done

**Review by daniellienert — APPROVED** [review]

**Pull request merged**